src/react-components/room/useAvatarVolume.js

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect, useState } from "react";
+import { getPlayerInfo } from "../../utils/component-utils";
+import {
+  calcLevel,
+  calcLevelStepDown,
+  calcLevelStepUp,
+  calcGainMultiplier,
+  MAX_VOLUME_LABELS
+} from "../../utils/avatar-volume-utils";
+
+const minLevel = 0;
+const maxLevel = MAX_VOLUME_LABELS - 1;
+
+export default function useAvatarVolume(sessionId) {
+  const playerInfo = getPlayerInfo(sessionId);
+  const controlsEl = playerInfo?.el.querySelector("[avatar-volume-controls]");
+  const controls = controlsEl?.components["avatar-volume-controls"];
+  const [levelStep, setLevelStep] = useState(0);
+  const [level, setLevel] = useState(0);
+  const [isMuted, setIsMuted] = useState(false);
+
+  const onGainMultiplierUpdated = useCallback(
+    ({ detail: { gainMultiplier } }) => {
+      const newLevel = calcLevel(gainMultiplier);
+      setLevelStep(newLevel > level ? calcLevelStepDown(gainMultiplier) : calcLevelStepUp(gainMultiplier));
+      setLevel(newLevel);
+    },
+    [level]
+  );
+
+  const updateGainMultiplier = useCallback(
+    value => {
+      if (!controls) return;
+      const gainMultiplier = calcGainMultiplier(value);
```

Comment:
Actually looks like what you pass in is indeed in "level" units, so the name is confusing. But then you basically end up having to convert back to gain multiplier, the actual value we want (and then converting back to level in onGainMultiplierUpdated)

Reply:
updated